sdcm/utils/sstable/load_utils.py

```diff
@@ -52,13 +57,21 @@ def distribute_test_files_to_cluster_nodes(cls, nodes, test_data: List[TestDataI
 
     @staticmethod
     def upload_sstables(node, test_data: TestDataInventory, keyspace_name: str = 'keyspace1',
-                        create_schema: bool = False, **kwargs):
+                        create_schema: bool = False, is_cloud_cluster=False, **kwargs):
         key_store = KeyStore()
         creds = key_store.get_scylladb_upload_credentials()
         # Download the sstable files from S3
-        remote_get_file(node.remoter, test_data.sstable_url, test_data.sstable_file,
-                        hash_expected=test_data.sstable_md5, retries=2,
-                        user_agent=creds['user_agent'])
+        if is_cloud_cluster:
+            local_temp_path = '/tmp/' + os.path.basename(test_data.sstable_file)
+            remote_get_file(LOCAL_CMD_RUNNER, test_data.sstable_url, local_temp_path,
+                            hash_expected=test_data.sstable_md5, retries=2,
+                            user_agent=creds['user_agent'])
+            node.remoter.send_files(local_temp_path, test_data.sstable_file)
+            LOCAL_CMD_RUNNER.run(f"rm -f {local_temp_path}", ignore_status=True)
```

Comment:
what if send_files would fail ? might consider putting this in a finally clause

Reply:
Done

Reply:
agree